src/torznab.ts

```diff
@@ -681,3 +662,76 @@ async function makeRequests(
 		candidates: results,
 	}));
 }
+async function getAndLogIndexers(
+	enabledIndexers: Indexer[],
+	timestampDataSql,
+	mediaType: MediaType,
+	searchee: Searchee,
+): Promise<{ indexersToUse: Indexer[]; parsedMedia?: ParsedMedia }> {
+	const { excludeRecentSearch, excludeOlder } = getRuntimeConfig();
+	const mediaTypeStr = mediaType.toUpperCase();
+
+	const indexersToUseByTimestamp = enabledIndexers.filter((indexer) => {
+		const entry = timestampDataSql.find(
+			(entry) => entry.indexerId === indexer.id,
+		);
+		if (!entry) {
+			return true;
+		}
+		return (
+			(!excludeOlder || entry.firstSearched > nMsAgo(excludeOlder)) &&
+			(!excludeRecentSearch ||
+				entry.lastSearched < nMsAgo(excludeRecentSearch))
+		);
+	});
+
+	const indexersToUse = indexersToUseByTimestamp.filter((indexer) => {
+		return indexerDoesSupportMediaType(
+			mediaType,
+			JSON.parse(indexer.categories),
+		);
+	});
+	const skippingSearching = (reason: string) =>
+		`Skipped searching on indexers for ${searchee.name} ${reason} [${mediaTypeStr}]`;
+
+	let skippingIndexersCallout =
+		enabledIndexers.length > indexersToUse.length ? " (filtered by " : "";
+
+	if (enabledIndexers.length > indexersToUseByTimestamp.length) {
+		skippingIndexersCallout +=
+			indexersToUseByTimestamp.length === indexersToUse.length
+				? "timestamps)"
+				: "timestamps ";
+	}
+	if (indexersToUseByTimestamp.length > indexersToUse.length) {
+		skippingIndexersCallout +=
+			enabledIndexers.length === indexersToUseByTimestamp.length
+				? "supported category)"
+				: "and supported category)";
```

Comment:
there's a util for this: https://github.com/cross-seed/cross-seed/blob/9558b31f908d731d184c2d2b17b983175aab1d69/src/utils.ts#L163